service/telemetry.go

```diff
@@ -173,8 +174,12 @@ func (tel *telemetryInitializer) initOpenCensus(cfg telemetry.Config, telAttrs m
 	}
 
 	// Until we can use a generic metrics exporter, default to Prometheus.
+	namespace := defaultNameSpace
+	if cfg.Metrics.NameSpace != "" {
```

Comment:
The [issue](https://github.com/open-telemetry/opentelemetry-collector/issues/5692) description said:
>If the user decides not to have any namespace to make the metrics more readable, it should be an option.

It doesn't seem to be possible to specify empty string as a namespace with this implementation. Is it a non-goal?

Reply:
I actually do want to support empty namespace. But I am also worried that will break a lot of existing users' metrics. As now their metrics will not have "otelcol" prefix by default. Will that kind of breaking changes be acceptable in general?

Reply:
No, we should not make breaking changes. "otelcol" must remain the default.

However, as one possible option, we can differentiate between unspecified `namespace` setting (meaning that default "otelcol" should be used) and explicitly specified empty string `namespace: ""`in the config. This can be done if `Namespace` is declared as `Namespace *string` optional setting in the `MetricsConfig`. I am not entirely sure this behavior is clear and unsurprising enough, perhaps there are other ways to allow specifying empty string, but also keep the current behavior as the default.

Reply:
Having a string pointer as config does seem a bit confusing. Another alternative is to use an additional bool field `UseCustomNamespace`. But then now we have two fields which increases maintenance effort...

I can do either. Do you have a preference?

Reply:
>Having a string pointer as config does seem a bit confusing.

IIRC using pointers is the idiomatic way to handle/detect optional config settings in Go yaml parsers. That itself should be fine and is not a problem in the code.

What I am more concerned about is whether having the default value of a string config setting as "otelcol" is a surprising behavior for the end users or we think it is fine. It is probably fine. I am curious what other @open-telemetry/collector-approvers think.

Reply:
ok if that is the convention. I can update to use the string pointer.

Regarding the default settings of "otelcol", I think that is ok as other otel collector also provide default configs or some sort. But if anyone from that list disagrees please comment on the thread.

Reply:
Let's see if anyone else objects and if not let's go with the pointer.